src/runtime_src/core/common/scheduler.cpp

```diff
@@ -184,8 +184,12 @@ init(xclDeviceHandle handle, const axlf* top)
       scmd->count = sizeof (ert_configure_sk_cmd) / 4 - 1;
       scmd->start_cuidx = start_cuidx;
       scmd->num_cus = sk.ninst;
-      std::strncpy(reinterpret_cast<char*>(scmd->sk_name), sk.symbol_name, 31);
-      reinterpret_cast<char*>(scmd->sk_name)[31] = 0;
+      
+      //std::strncpy(reinterpret_cast<char*>(scmd->sk_name), sk.symbol_name, 31);
+      memset(reinterpret_cast<char*>(scmd->sk_name), 0, 32);
+      sk.symbol_name.copy(reinterpret_cast<char*>(scmd->sk_name), 31);
+      //reinterpret_cast<char*>(scmd->sk_name)[31] = 0;
```

Comment:
Not understanding this?  Why do we need to write to sk_name twice?
Please always remove dead code, e.g. remove the commented out lines, the git history has it.

Reply:
removed. memset 0 of full data structure scmd occurred earlier